docs/examples/substrate/call_flags.sol

```diff
@@ -0,0 +1,23 @@
+contract CallFlags {
+
+    enum CallFlag { FORWARD_INPUT, CLONE_INPUT, TAIL_CALL, ALLOW_REENTRY }
+
+    // Set all flags found in _flags to true.
+    function bitflags(CallFlag[] _flags) internal pure returns (uint32 flags) {
+        for (uint n = 0; n < _flags.length; n++) {
+            flags |= (2 ** uint32(_flags[n]));
+        }
+    }
```

Comment:
This is not very readable, and it's part of the documentation. How about:

```solidity
uint32 constant FORWARD_INPUT = 1;
uint32 constant CLONE_INPUT = 2;
uint32 constant TAIL_CALL = 4;
uint32 constant ALLOW_REENTRY = 8;
```
You could even put it in a library:

```solidity
library CallFlags {
uint32 constant FORWARD_INPUT = 1;
uint32 constant CLONE_INPUT = 2;
uint32 constant TAIL_CALL = 4;
uint32 constant ALLOW_REENTRY = 8;
}

contract c {
   function call_with_flags(
        address _address,
        bytes4 _selector,
    ) public returns (bytes ret) {
        uint32 call_flags = bitflags(_flags);
        (bool ok, ret) = _address.call{flags: CallFlags.ALLOW_REENTRANT}(_selector);
        require(ok);
    }
}
```

It could also go into `substrate-library/`.

Reply:
I like that!

Reply:
I changed the example to a much simpler and more to the point version.

Gonna introduce the `substrate-library/` in a later PR. The enum version, while a bit less readable is really nice for end-users. But the constants are nice to have as well. I think we want both. Additionally, I think the `Hash` type from the `substrate` builtin could also be shoved off into a Solidity version. But it's all out of scope for this PR.